dune-project

```diff
@@ -1,2 +1,43 @@
 (lang dune 3.16)
-(using menhir 2.0)
+
+(name hazel)
+
+(generate_opam_files true)
+
+(source
+ (github hazelgrove/hazel))
+
+(authors "Hazel Development Team")
+(maintainers "Hazel Development Team")
+
+(license MIT)
+
+(package
+ (name hazel)
+ (allow_empty)
+ (synopsis "Hazel, a live functional programming environment with typed holes")
+;  (description "A longer description")
+ ; (tags
+ ; (topics "to describe" your project))
+ (depends
+  (ocaml
+   (>= 5.2.0))
+  (menhir
+   (>= 2.0))
+  yojson
+  reason
+  ppx_yojson_conv_lib
+  ppx_yojson_conv
+  incr_dom
+  (omd (>= 2.0.0~alpha4))
+  ezjs_idb
+  virtual_dom
+  ppx_deriving
+  ptmap
+  uuidm
+  unionFind
+  ocamlformat
+  (junit_alcotest :with-test)
+  ocaml-lsp-server)) ; After upgrading to opam 2.2 use with-dev https://opam.ocaml.org/blog/opam-2-2-0/
```

Comment:
We'll separate dev dependencies when we go to opam 2.2.0

Reply:
Is there a reason not to go to opam 2.2 now?

Reply:
I just didn't want to mix this with any upgrades in case something breaks so we can individually revert

Reply:
Okay, let's go ahead and do that in a separate PR after this one then.

Reply:
Working on that PR now. Taking a second look we all install opam system-wide from the OS so we probably want to wait until 2.2 is widely distributed by distributions https://repology.org/project/opam/versions. Ubuntu/Nix are still on 2.1. Homebrew is on 2.2.

Reply:
The dune project also needs to require higher versions if we want the lock file to explicitly require 2.2 https://github.com/ocaml/dune/blob/d5809e9de4480a904ca341cbe062880648d8fa5e/src/dune_rules/opam_create.ml#L242

Reply:
@7h3kk1d can you make a tracking issue for this

Reply:
https://github.com/hazelgrove/hazel/issues/1365